### PR TITLE
Update ClickHouse driver to 0.3.2-patch1

### PR DESCRIPTION
**Pull request by wendigo**

### CODE REVIEW

**Review by grantatspothero — APPROVED** [review]
LGTM thanks for doing this, glad to see a new [clickhouse-jdbc version ](https://github.com/ClickHouse/clickhouse-jdbc/tags)was released last week 🎉

File: pom.xml [hunk 1]
```diff
@@ -981,7 +981,7 @@
             <dependency>
                 <groupId>com.github.ben-manes.caffeine</groupId>
                 <artifactId>caffeine</artifactId>
-                <version>3.0.3</version>
+                <version>3.0.4</version>
```
[the remark this answers is not in the document]
Comment (wendigo) [reply pom.xml:984]: Yes.

**Review by hashhar — APPROVED** [review]

**Review by ebyhr — COMMENTED** [review]
Please run stress tests (~300 attempts) for #9300. When I was testing #9350, I faced the different flaky message after upgrade.

**Review by ebyhr — COMMENTED** [review]
New version started supporting `LocalDateTime` as far as I confirmed. It would be nice to fix below mapping. Another PR is fine. 
https://github.com/trinodb/trino/blob/master/plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java#L435-L438

**Comment by wendigo** [comment]
@ebyhr @hashhar PTAL and merge if you're okay with this change

**Comment by grantatspothero** [comment]
@wendigo I noticed clickhouse just released a `0.3.2-patch1` version to affect an unrelated datetime/timestamp bug:
https://github.com/ClickHouse/clickhouse-jdbc/releases/tag/v0.3.2-patch1

Probably worth bumping to the patched version

**Comment by wendigo** [comment]
@grantatspothero bumped

**Comment by ebyhr** [comment]
@grantatspothero We won't meet the NPE condition even in 0.3.2 because our `ReadFunction.isNull()` skips null using `ResultSet.getObject()` (not using `getDate`).

**Review by ebyhr — APPROVED** [review]

**Comment by ebyhr** [comment]
Merged, thanks!

**Pull request merged**